src/uct/ib/rc/base/rc_iface.h

```diff
@@ -126,9 +126,14 @@ typedef struct uct_rc_hdr {
 } UCS_S_PACKED uct_rc_hdr_t;
 
 
-typedef struct uct_rc_fc_request {
+typedef struct uct_rc_pending_req {
     uct_pending_req_t super;
     uct_ep_t          *ep;
+} uct_rc_pending_req_t;
+
+
+typedef struct uct_rc_fc_request {
+    uct_rc_pending_req_t super;
```

Comment:
the problem is that we create the mpool with elements of same size, so better not use inheritance here (to prevent case where the mpool object size is too small)
IMO should use union, similarly to UCP requests for example, or rc outstanding operations